site-packages/integralstor_gridcell/grid_ops.py

```diff
@@ -980,6 +984,33 @@ def add_a_gridcell_to_storage_pool(si, hostname):
             raise Exception(','.join(error_list))
         print 'Mounted admin volume.'
 
+        # Check if the admin vol mount actually succeded by reading a file from the admin volume
+        # Tries 3 times. Bail out if unsuccessfull
+        print "Check if mounted admin vol is accessible"
+        if rc is True:
+            iter = 3
+            while (iter):
+                iter -= 1
+                is_mount_ok = True
+                ret = client.cmd(hostname, 'file.file_exists', [
+                    '/opt/integralstor/integralstor_gridcell/config/lock/assert_admin_vol_mount'])
+                if ret:
+                    for node, val in ret.items():
+                        if val is False:
+                            print "Mounted vol not readable! Retrying after 5 seconds."
+                            is_mount_ok = False
+                            break
+
+                    if is_mount_ok is True:
```

Comment:
I have accounted for that. It's alright if we end up with `is_mount_ok` being `True`: the `if iter < 1` check is the one which raises the exception. Loop runs thrice even if `ret.items()` doesn't carry anything.